### PR TITLE
Fix templates sourceName & port of template fixes from main

### LINKED ISSUE #41519: Backport template fixes to 6.0.xxx
Backport #41499 to 6.0.xxx

### PR DESCRIPTION
**Pull request by DamianEdwards**
This PR fixes two template issues that were introduced by the recent addition of the `--use-program-main` option, as well as another issue discovered during implementation of the fixes:

1. #41529
    ## Description
    In the Blazor WebAssembly, Blazor Server, and gRPC project templates, if the `--use-program-main` option is specified, the created `Program.cs` file contains a namespace declaration that does not match the project name, but instead is always `Company.WebApplication1`. 

    ## Customer Impact
    This doesn't result in any compiler warning or other build issues on its own, but the name is obviously wrong.

   ## Cause
    This is due to the namespaces not being updated for each specific project template when it was copy/pasted on authoring, as each project template uses a different placeholder namespace that is then replaced with the project name after creation.

    ## Discovered
    This issue was found internally during verification of 6.0.300. The fix has already been [merged to main](https://github.com/dotnet/aspnetcore/pull/41499).

    ## Testing
    Automated tests have been updated to verify that the declared namespaces in the generated *.cs files start with the specified project name. The changes were also verified with manual testing.

    ## Regression
    This is not a regression as it's an issue in the implementation of a newly added feature.

1. #41491
    ## Description
    In the Web API project template, if both the `--use-program-main` and `--use-minimal-apis` options are specified, the project ends up with two different entry points: one in `Program.cs` and another in `Program.Main.cs`.

    ## Customer Impact
    This results in a compiler warning when trying to build the created project:
    ```
    >dotnet build
    Microsoft (R) Build Engine version 17.2.0+41abc5629 for .NET
    Copyright (C) Microsoft Corporation. All rights reserved.
    
      Determining projects to restore...
      All projects are up-to-date for restore.
    C:\Users\<user>\Solutions\Projects02\WebApi02\Program.Main.cs(5,24): warning CS7022: The entry point of the program is
    global code; ignoring 'Program.Main(string[])' entry point. [C:\Users\<user>\Solutions\Projects02\WebApi02\WebApi02.csp
    roj]
      WebApi02 -> C:\Users\<user>\Solutions\Projects02\WebApi02\bin\Debug\net6.0\WebApi02.dll
    
    Build succeeded.
    
    C:\Users\<user>\Solutions\Projects02\WebApi02\Program.Main.cs(5,24): warning CS7022: The entry point of the program is
    global code; ignoring 'Program.Main(string[])' entry point. [C:\Users\<user>\Solutions\Projects02\WebApi02\WebApi02.csp
    roj]
        1 Warning(s)
        0 Error(s)
    
    Time Elapsed 00:00:05.16
    ```
    ## Cause
    This is due to an authoring bug in the template config that doesn't correctly handle the case when both options are specified.

    ## Discovered
    This issue was found internally during verification of 6.0.300. The fix has already been [merged to main](https://github.com/dotnet/aspnetcore/pull/41499).

    ## Testing
    Automated tests have been updated to verify that only the expected files are generated for all template option combinations, along with verifying that no compiler warnings are emitted when building newly created projects. The changes were also verified with manual testing.

    ## Regression
    This is not a regression as it's an issue in the implementation of a newly added feature.

1. **Web API template produces unbuildable project when `--use-minimal-apis` and any authentication (`-au`) option is supplied.**
    ## Description
    In the Web API project template, if both the `--use-minimal-apis` and any cloud authentication (`-au`) option is supplied, the resultant project contains numerous code errors that result in build failures. This issue only manifests when using the templates from the command line (`dotnet new`) as Visual Studio uses its own mechanism for configuring authentication in new projects.

    ## Customer Impact
    New minimal API projects with cloud authentication options created from the command line cannot be built.

   ## Cause
    This is due to authoring mistakes in the template itself.

    ## Discovered
    These issues were discovered as part of reviewing the templates when implementing the fixes for the issues described above. Despite these mistakes being in the templates since the `--use-minimal-apis` option was added during 6.0.0, there were no customer reports. The fix has already been [merged to main](https://github.com/dotnet/aspnetcore/pull/41499).

    ## Testing
    Automated tests have been updated to verify that the declared namespaces in the generated *.cs files start with the specified project name. The changes were also verified with manual testing.

    ## Regression
    This is not a regression as it's an issue in the implementation since the feature was added.

Addresses #41529 in 6.0.xxx
Fixes #41519

### CODE REVIEW

**Comment by DamianEdwards** [comment]
This has been approved by shiproom for 6.0.6 and is ready for review @HaoK @BrennanConroy @dougbu @wtgodbe

File: eng/helix/content/RunTests/TestRunner.cs [hunk 1]
```diff
@@ -213,50 +213,60 @@ public async Task<int> RunTestsAsync()
             try
             {
                 // Timeout test run 5 minutes before the Helix job would timeout
-                var cts = new CancellationTokenSource(Options.Timeout.Subtract(TimeSpan.FromMinutes(5)));
+                var testProcessTimeout = Options.Timeout.Subtract(TimeSpan.FromMinutes(5));
+                var cts = new CancellationTokenSource(testProcessTimeout);
                 var diagLog = Path.Combine(Environment.GetEnvironmentVariable("HELIX_WORKITEM_UPLOAD_ROOT"), "vstest.log");
-                var commonTestArgs = $"test {Options.Target} --diag:{diagLog} --logger:xunit --logger:\"console;verbosity=normal\" --blame \"CollectHangDump;TestTimeout=15m\"";
+                var commonTestArgs = $"test {Options.Target} --diag:{diagLog} --logger:xunit --logger:\"console;verbosity=normal\" --blame \"CollectHangDump;CollectDump;TestTimeout=15m\"";
```
Comment (HaoK) [inline eng/helix/content/RunTests/TestRunner.cs:219]: Do we want this to use the same (new syntax) arguments as 7.0?

File: eng/helix/content/RunTests/TestRunner.cs [hunk 2]
```diff
@@ -213,50 +213,60 @@ public async Task<int> RunTestsAsync()
...
-                        errorDataReceived: Console.Error.WriteLine,
+                        outputDataReceived: ProcessUtil.PrintMessage,
+                        errorDataReceived: ProcessUtil.PrintErrorMessage,
                         throwOnError: false,
                         cancellationToken: cts.Token);
 
                     if (result.ExitCode != 0)
                     {
-                        Console.WriteLine($"Failure in quarantined tests. Exit code: {result.ExitCode}.");
+                        if (cts.Token.IsCancellationRequested)
```
Comment (HaoK) [inline eng/helix/content/RunTests/TestRunner.cs:235]: I think this is also missing the change that moved this outside of the exit code check: https://github.com/dotnet/aspnetcore/commit/cab4c6ec6e900b2c4b5b93f2237c6972efebc02a#diff-bf8c10d051da7b9b3b26d56a968fe4ea30048d1da46a7474c75dd089c50bc1ffR241

File: eng/helix/content/RunTests/TestRunner.cs (hunk 1 above)
Comment (DamianEdwards) [reply eng/helix/content/RunTests/TestRunner.cs:219]: May as well 👍

**Review by HaoK — APPROVED** [review]

File: eng/helix/content/RunTests/TestRunner.cs [hunk 3]
```diff
@@ -213,50 +213,60 @@ public async Task<int> RunTestsAsync()
             try
             {
                 // Timeout test run 5 minutes before the Helix job would timeout
-                var cts = new CancellationTokenSource(Options.Timeout.Subtract(TimeSpan.FromMinutes(5)));
+                var testProcessTimeout = Options.Timeout.Subtract(TimeSpan.FromMinutes(5));
+                var cts = new CancellationTokenSource(testProcessTimeout);
                 var diagLog = Path.Combine(Environment.GetEnvironmentVariable("HELIX_WORKITEM_UPLOAD_ROOT"), "vstest.log");
-                var commonTestArgs = $"test {Options.Target} --diag:{diagLog} --logger:xunit --logger:\"console;verbosity=normal\" --blame \"CollectHangDump;TestTimeout=15m\"";
+                var commonTestArgs = $"test {Options.Target} --diag:{diagLog} --logger xunit --logger \"console;verbosity=normal\" " +
+                                    "--blame-crash --blame-hang-timeout 15m";
```
Comment (dougbu) [inline eng/helix/content/RunTests/TestRunner.cs:220]: Does `--blame-hang-timeout` imply `--blame-hang` @nohwnd  @MarcoRossignoli❔

**Review by dougbu — APPROVED** [review]

File: eng/helix/content/RunTests/TestRunner.cs (hunk 3 above)
Comment (nohwnd) [reply eng/helix/content/RunTests/TestRunner.cs:220]: Going through all my missed notifications. Yes it does. It enables --blame (creating the sequence xml file) and hang dumps with the default dump type (full), and non-default timeout (what you define for the --blame-hang-timeout). Same goes for ---blame-hang-dump-type, or --blame-crash-dump-type implying --blame-crash.

**Pull request merged**